### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by francisbouchard**
Authentication uses express session.

### CODE REVIEW

**Comment by adriannadiaz** [comment]
Also, without being logged in,  I write 

http://localhost:4200/profiles

and I am able to get every participant's information

File: angular-client/src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 })
 export class AppComponent {
+  constructor(
+    private authenticationService: AuthenticationService,
+    private router: Router
+  ){}
   title = 'app';
+  public logout(){
+    this.authenticationService.logout().subscribe(data => {
+      this.router.navigateByUrl('/login')
```
Comment (Ldhatter) [inline angular-client/src/app/app.component.ts]: it doesnt go to this page.
screen shot 2018-01-13 at 10 41 49 am

File: angular-client/src/app/app.component.html [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+    <a class="nav-link" routerLink="new-participant">
+      <button mat-button>New Participant</button>
+    </a>
+    <a class="nav-link" routerLink="profiles">
+      <button mat-button>Profiles</button>
+    </a>
+  </span>
+  <span class="example-fill-remaining-space"></span>
+  <span>
+    <button mat-raised-button color="warn" (click)="logout()">Logout</button>
```
Comment (Ldhatter) [inline angular-client/src/app/app.component.html]: when we launch application, button does not do anything. so i think we need to make sure that the first page is the login or signup page

File: angular-client/src/app/app.component.ts (hunk 1 above)
Comment (Ldhatter) [reply angular-client/src/app/app.component.ts]: i think we need to make sure that the first page we see is the login page

File: angular-client/src/app/components/login/login.component.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <mat-spinner *ngIf="loading"></mat-spinner>
+      <div *ngIf="error">
+          <i class="fa fa-exclamation" aria-hidden="true"></i>
+          <mat-icon>error</mat-icon>
+          <h4>{{msg}}</h4>
+      </div>
+      
+    </mat-card>
+  </div>
+</div>
```
Comment (Ldhatter) [inline angular-client/src/app/components/login/login.component.html]: need to implement a signup options

File: angular-client/src/app/services/authentication.service.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+@Injectable()
+export class AuthenticationService {
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  public login(email: string, password: string) {
+    return this.http.post<any>('/user/login', { email: email, password: password })
+  }
```
Comment (Ldhatter) [inline angular-client/src/app/services/authentication.service.ts]: needs to catchError

File: angular-client/src/app/services/authentication.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  public logout() {
+    return this.http.post<any>('/user/logout', {})
+  }
```
Comment (Ldhatter) [inline angular-client/src/app/services/authentication.service.ts]: pipe a catchError

File: angular-client/src/app/services/authentication.service.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  public signup (email: string, password: string, confirmPassword: string) {
+    return this.http.post<any>('/user/signup', {})
+  }
```
Comment (Ldhatter) [inline angular-client/src/app/services/authentication.service.ts]: pipe a catchError

File: angular-client/src/app/services/authentication.service.ts (hunk 5 above)
Comment (francisbouchard) [reply angular-client/src/app/services/authentication.service.ts]: This just returns the promise, the error gets caught when used

File: angular-client/src/app/services/authentication.service.ts (hunk 6 above)
Comment (francisbouchard) [reply angular-client/src/app/services/authentication.service.ts]: ^

File: angular-client/src/app/components/login/login.component.html (hunk 3 above)
Comment (francisbouchard) [reply angular-client/src/app/components/login/login.component.html]: How are we implementing signup? Can anyone signup to this system?

**Pull request merged**